src/templates/next/components/Button/Button.tsx

```diff
@@ -1,45 +1,120 @@
 import { ButtonProps } from "~/common"
-import { NextButtonColorVariant } from "~/common/Button"
+import { NextButtonColorScheme } from "~/common/Button"
 import { SUPPORTED_ICONS_MAP } from "~/common/Icons"
 
-const colorVariantToClassMap: Record<NextButtonColorVariant, string> = {
-  black: "bg-[#333333] text-white",
-  white: "bg-white text-[#333333]",
-}
-
-const Button = ({ label, colorVariant, href, rightIcon }: ButtonProps) => {
-  const Label = () => (
-    <span className="text-center text-lg font-semibold leading-tight">
-      {label}
-    </span>
-  )
+const Label = ({ label }: Pick<ButtonProps, "label">) => (
+  <span className="text-center text-lg font-semibold leading-tight">
+    {label}
+  </span>
+)
 
-  const RightIcon = () => {
-    if (!rightIcon) {
-      return null
-    }
-    const Icon = SUPPORTED_ICONS_MAP[rightIcon]
-    return (
-      <div>
-        <Icon className="size-6" />
-      </div>
-    )
+const RightIcon = ({ rightIcon }: Pick<ButtonProps, "rightIcon">) => {
+  if (!rightIcon) {
+    return null
   }
+  const Icon = SUPPORTED_ICONS_MAP[rightIcon]
+  return (
+    <div>
+      <Icon className="size-6" />
+    </div>
+  )
+}
 
-  const colorVariantClass = colorVariantToClassMap[colorVariant ?? "black"]
-
+const BaseButton = ({
+  label,
+  href,
+  rightIcon,
+  className = "",
+  isLinkVariant = false,
+}: ButtonProps & { className?: string; isLinkVariant?: boolean }) => {
   return (
     <a
```

Comment:
not part of this pr but the base button is an `a`?

Reply:
yep, because semantically they are all links

Reply:
Kek

Reply:
This effectively blocks us from having actual buttons - if that's possible on next, then this decision needs to be well thought through.

Reply:
fair, but I think it'll be quite easy to just differentiate them in the future if needed
this decision now doesn't really lock us into anything long term